core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -114,10 +114,10 @@ private[spark] abstract class LauncherBackend {
 
     override def close(): Unit = {
       try {
+        _isConnected = false
         super.close()
       } finally {
         onDisconnected()
```

Comment:
I searched the code and seems this is a no-op?

Reply:
`_isConnected` is used in `def isConnected()`. Moving it from the `finally` block to before the call to `super.close()` avoids a race condition where a client tries to write to the connection after it has been closed.

Reply:
I know your meaning, but I'm not referring to `isConnected`, I mean `onDisconnected` seems not doing anything on Spark side.

Reply:
This is not quite directly related, but this should means it is safe to move `_isConnected = false` to before `onDisconnected()`.

Reply:
Yeah, your right, it doesn't look like `onDisconnected` is being used anywhere, but its marked as `protected` so I guess its meant to be used by a sub-class (although no sub-class uses it).

So agree, this change should be safe, it doesn't change the semantics of any use of `onDisconnected`